### PR TITLE
Fix issue #20 "unexpected end of JSON input" error

### PR DESCRIPTION
**Pull request by mdb**
This fixes issue #20 by correcting the `reader.CreateReader` logic determining whether to use a `reader.FileReader` or `reader.StdinReader`. Previously, in some environments (like GH Actions Ubuntu runners), `tf-summarize` incorrectly attempted to use a `reader.StdinReader`, even when an explicit filename argument is provided.

Ideally, `tf-summarize` would feature automated tests through which I could prove the bug & fix. ~In absence of such tests,~ I added a `demo` GitHub Actions job proving `tf-summarize` continues to work. ~However, I'm happy to add real tests (either in this PR, or in a followup PR)~, if you prefer those, @dineshba (I'm also happy to remove the `demo` GH Actions job from this PR if you prefer it not clutter `.github/workflows/build.yml`).

**Update:** I added 2 additional commits:

* https://github.com/dineshba/tf-summarize/pull/64/commits/40f8d1bd71529351c5b051a032fc5f3d646f0960 ensures the use of a consistent Terraform version when interacting with the `example` project
* https://github.com/dineshba/tf-summarize/pull/64/commits/efdf4dbd5ff3364c0f39ed0f6bbb70c7ecba92df adds an automated test verifying the fix provided by this PR showing that `tf-summarize -md plan.json` and `cat plan.json | tf-summarize -md` both continue to work as expected.

Note that https://github.com/dineshba/tf-summarize/pull/63 shows reproduction of the bug via a similar GitHub Actions-based demo.

### CODE REVIEW

File: reader/reader.go [hunk 1]
```diff
@@ -22,19 +23,22 @@ func readFile(f io.Reader) ([]byte, error) {
...
+	if len(input) == 0 {
+		return nil, errors.New("no input data; expected input via a non-empty file or via STDIN")
 	}
 	return input, nil
 }
 
-func CreateReader(stdin *os.File, args []string) (Reader, error) {
-	stat, _ := stdin.Stat()
-	if (stat.Mode() & os.ModeCharDevice) == 0 {
-		return NewStdinReader(), nil
```
Comment (mdb) [inline reader/reader.go:33]: ☝️ Previously, in some environments -- like GitHub Actions Ubuntu runners -- `tf-summarize` enters this codepath, even when an explicit filename argument has been provided.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -2,34 +2,89 @@ name: Build
...
+        with:
+          version: latest
+
+      - name: Test
+        run: go test -v ./...
+
+      - name: Build
+        run: go build
+
+  demo:
```
Comment (mdb) [inline .github/workflows/build.yml:36]: As I mention in the main PR comment, I'm happy to remove this job if you'd prefer it not cluster the `build.yml` workflow.

**Comment by mdb** [comment]
👋 Hi @dineshba - Have you been able to take a look at this? I'm curious to hear your thoughts. Thanks!

**Review by dineshba — CHANGES_REQUESTED** [review]
Can we add tfplan, tfplan.json in example folder to .gitignore?

File: reader/reader.go (hunk 1 above)
Comment (dineshba) [reply reader/reader.go:33]: Got it. Resolving

File: .github/workflows/build.yml (hunk 2 above)
Comment (dineshba) [reply .github/workflows/build.yml:36]: I would like to keep it in separate file

**Comment by dineshba** [comment]
Other than this, all changes good. Really appreciate the effort and the fix 👌

File: .github/workflows/build.yml (hunk 2 above)
Comment (mdb) [reply .github/workflows/build.yml]: Sounds good; I've removed it!

**Comment by mdb** [comment]
> Can we add tfplan in example folder to .gitignore?

@dineshba Good call; I've removed the `example/tfplan` file and added it to the `.gitignore` (I've left `example/tfplan.json` in place, as this file is used by the automated tests).

**Review by dineshba — APPROVED** [review]

**Pull request merged**